scadalts-ui/src/views/DataObjects/Scripting/index.vue

```diff
@@ -1,33 +1,420 @@
+			if (oc && oc.length) {
+				const o1 = oc.find((x) => x.key == 1);
+				const o2 = oc.find((x) => x.key == 2);
+				if (o1 && o1.value) {
+					this.scriptForm.datasourceContext = o1.value;
+				} else {
+					this.scriptForm.datasourceContext = '';
+				}
+
+				if (o2 && o2.value) {
+					this.scriptForm.datapointContext = o2.value;
+				} else {
+					this.scriptForm.datapointContext = '';
+				}
+
+				this.scriptForm.datapointContext = o2.value;
+			} else {
+				this.scriptForm.datasourceContext = '';
+				this.scriptForm.datapointContext = '';
+			}
+
+			this.scriptForm.script = this.selectedScript.script;
+		},
+		addDatapoint() {
+			const p = this.datapoints.find((x) => x.id === this.selectedDatapointId);
+			if (!this.scriptForm.pointsOnContext.find((x) => p.xid === x.dataPointXid)) {
+				this.scriptForm.pointsOnContext.push({
+					dataPointXid: p.xid,
+					varName: `p${p.id}`,
+				});
+				this.selectedDatapointId = null;
+			}
+		},
+		async fetchScriptList() {
+			this.loading = true;
+			this.scriptList = await this.$store.dispatch('searchScripts', this.searchFilters);
+			if (!this.search) {
+				this.scriptListFiltered = this.scriptList;
+			} else {
+				const keywords = this.search.split(' ');
+				this.scriptListFiltered = this.scriptList.filter((x) =>
+					`${x.id} ${x.xid} ${x.name} ${x.script}`
+						.toLowerCase()
+						.includes(keywords[0].toLowerCase()),
+				);
+			}
+			this.totalScripts = this.scriptList.total;
+		},
+		async runScript(xid) {
+			try {
+				await this.$store.dispatch('runScript', xid);
+				this.snackbar = true;
+				this.snackbarMessage = `${this.$t('scriptList.successfulScriptExecution')} `;
+			} catch (e) {
+				this.snackbar = true;
+				this.snackbarMessage = `${this.$t('scriptList.failedScriptExecution')} `;
+			}	
+		},
+		async saveAndRunScript(xid) {
+			await this.saveScript(false)
+			try {
+				await this.$store.dispatch('runScript', xid);
+				this.snackbar = true;
+				this.snackbarMessage = `${this.$t('scriptList.successfulScriptExecution')} `;
+			} catch (e) {
+				this.snackbar = true;
+				this.snackbarMessage = `${this.$t('scriptList.failedScriptExecution')} `;
+			}	
+		},
+		
+		async saveScript(closeOnSaveConfirmation = true) {
+			if (this.$refs.editForm.validate()) {
+				
+				let response
+				if (this.selectedScriptId != -1) {
+					response = await this.$store.dispatch('updateScript', this.scriptForm);
+				} else {
+					response = await this.$store.dispatch('createScript', this.scriptForm);
+				}
+				this.fetchScriptList()
+				this.snackbar = true;
+				this.snackbarMessage = `${this.$t('scriptList.scriptSaved')} `;
+				if (closeOnSaveConfirmation) this.dialog = false
+			}
+            
+		},
+		async deleteScript(id) {
+			this.scriptList = await this.$store.dispatch('deleteScript', id);
+			this.fetchScriptList();
+			this.dialog = false;
+			this.snackbar = true;
+			this.snackbarMessage = `${this.$t('scriptList.deletedScript')} #${id}`;
+		},
```

Comment:
This works great but what do you think about adding the "Confirmation Dialog"? Just before deletion of the Script from the dialog component or directly from the list. It user should be aware of his actions.